### PR TITLE
feat: add menu dumping

### PR DESCRIPTION
**Pull request by Lierrmm**

### CODE REVIEW

File: src/component/assets/Menus.cpp [hunk 1]
```diff
@@ -0,0 +1,1089 @@
...
+
+						auto header = game::DB_FindXAssetHeader(game::qos::ASSET_TYPE_MENULIST, name);
+						if (!header.menuList)
+						{
+							console::error("dumpmenulist failed on '%s'\n", name);
+							return;
+						}
+
+						/*auto converted = menus::convert(header.menu);
+						map_dumper::api->write(game::iw4::ASSET_TYPE_MENU, converted);*/
```
Comment (mjkzy) [inline src/component/assets/Menus.cpp:1065]: writing the output disabled?

File: src/component/assets/Menus.cpp [hunk 2]
```diff
@@ -0,0 +1,1089 @@
...
+
+					auto header = game::DB_FindXAssetHeader(game::qos::ASSET_TYPE_MENU, name);
+					if (!header.menu)
+					{
+						console::error("dumpmenu failed on '%s'\n", name);
+						return;
+					}
+
+					/*auto converted = menus::convert(header.menu);
+					map_dumper::api->write(game::iw4::ASSET_TYPE_MENU, converted);*/
```
Comment (mjkzy) [inline src/component/assets/Menus.cpp:1040]: same here?

File: src/component/assets/Menus.cpp [hunk 3]
```diff
@@ -0,0 +1,1089 @@
...
+
+			menu_file_ << "{\n";
+
+			write_menu_def(menu);
+
+			menu_file_ << "}\n";
+			menu_file_.close();
+		}
+
+		game::qos::menuDef_t* convert(game::qos::menuDef_t* menufile)
```
Comment (mjkzy) [inline src/component/assets/Menus.cpp:1001]: no convert code for menuDef_t?

**Review by mjkzy — CHANGES_REQUESTED** [review]
im gonna assume this is mainly for dumping. maybe we should rework the base to be `convert` & `dump` commands instead lol. anyways, do we want to add convert support to IW4 menus? doesn't seem too important to me at this time to do that though

**Comment by Lierrmm** [comment]
iw4-open-formats doesn't have a handler for converting menudefs, which is why it isn't complete in this commit.

**Comment by mjkzy** [comment]
will merge this from now, but probably should refactor later and add commands for convert & dumping

**Pull request merged**